src/UnitTests/GitHub.App/ViewModels/PullRequestCreationViewModelTests.cs

```diff
@@ -61,4 +61,72 @@ public void TemplateIsUsedIfPresent()
 
         Assert.Equal("Test PR template", vm.Description);
     }
+
+
+    [Fact]
+    public async Task FormIsEmptyIfCanceled()
+    {
+        var serviceProvider = Substitutes.ServiceProvider;
+        var service = serviceProvider.GetPullRequestsService();
+        var notifications = Substitute.For<INotificationService>();
+
+        var gitService = serviceProvider.GetGitService();
+        var repo = Substitute.For<IRepository>();
+        repo.Head.Returns(Substitute.For<Branch>());
+        gitService.GetRepo(Arg.Any<string>()).Returns(repo);
+
+        var host = serviceProvider.GetRepositoryHosts().GitHubHost;
+        var ms = Substitute.For<IModelService>();
+        var master = Substitute.For<IBranch>();
+        master.Name.Returns("master");
+        ms.GetBranches(Arg.Any<ISimpleRepositoryModel>()).Returns(Observable.Return(master));
+        host.ModelService.Returns(ms);
+
+        var repository = new SimpleRepositoryModel("name", new GitHub.Primitives.UriString("http://github.com/github/stuff"));
+
+        var vm = new PullRequestCreationViewModel(host, repository, service, notifications);
+        vm.PRTitle = "PR title";
+        vm.Description = "PR Desc";
+        vm.Initialize(null);
```

Comment:
You probably want to set the TargetBranch to something different here, so you can test whether it actually resets or not. That means returning two IBranch objects in the observable in the `GetBranches` call above.

Also, `PRTitle` and `Description` should be set after the `Initialize()` call (which doesn't need a null parameter, it has a default value). Users can only interact with the form after the `Initialize()` call occurs, so setting those properties after is closer to real world usage.